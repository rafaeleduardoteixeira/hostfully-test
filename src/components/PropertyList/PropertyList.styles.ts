import styled from 'styled-components';
import { colors, media } from '../../themes';

export const Container = styled.div`
  display: flex;
  flex-direction: column;

  justify-content: center;
  width: 100%;
  align-items: center;
  padding: 70px 0;
`;

export const PropertyListTitle = styled.h1`
  display: flex;
  justify-content: center;
  font-size: 2.5rem;
  font-weight: 400;
  color: ${colors.black};

  @media ${media.mobile} {
    font-size: 2rem;
  }
`;

export const Delimiter = styled.div`
  margin: 20px auto;
  background-position: center;
  background-image: url('/images/delimiter.png');
  height: 70px;
  width: 500px;

  @media ${media.mobile} {
    width: 100%;
    height: 50px;
  }
`;

export const PropertyCardContainer = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  gap: 30px;

  @media ${media.notebook} {
    grid-template-columns: 1fr 1fr 1fr;
  }

  @media ${media.tablet} {
    grid-template-columns: 1fr 1fr;
  }

  @media ${media.mobile} {
    grid-template-columns: 1fr;
  }
`;
