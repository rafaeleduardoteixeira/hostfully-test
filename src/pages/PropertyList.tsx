const PropertyList = (): JSX.Element => {
  return <div></div>;
};

export default PropertyList;
