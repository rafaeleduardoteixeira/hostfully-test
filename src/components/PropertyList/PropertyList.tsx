// Created a component called PropertyList that takes in isLoading and properties as props.
// It's the property list container that will display all the properties from props.

import { useEffect, useState } from 'react';
import { Container, Delimiter, PropertyListTitle, PropertyCardContainer } from './PropertyList.styles';
import { Property } from '../../interfaces/property';
import { PropertyCardSkeleton } from '../PropertyCardSkeleton/PropertyCardSkeleton';
import { PropertyCard } from '../PropertyCard/PropertyCard';

interface PropertyListProps {
  isLoading: boolean;
  properties: Property[] | null;
}

export const PropertyList = ({ isLoading, properties }: PropertyListProps): JSX.Element => {
  return (
    <Container>
      <PropertyListTitle>Properties For Rent</PropertyListTitle>
      <Delimiter />
      <PropertyCardContainer>
        {isLoading ? (
          <>
            <PropertyCardSkeleton />
            <PropertyCardSkeleton />
            <PropertyCardSkeleton />
            <PropertyCardSkeleton />
          </>
        ) : (
          properties?.map((property) => <PropertyCard key={property.id} property={property} />)
        )}
      </PropertyCardContainer>
    </Container>
  );
};
