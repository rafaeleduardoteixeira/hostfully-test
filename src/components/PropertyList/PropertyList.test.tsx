import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { PropertyList } from './PropertyList';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useNavigate: () => mockedUsedNavigate,
}));

const properties = [
  {
    id: '1',
    name: 'Property 1',
    description: 'Description 1',
    city: 'City 1',
    country: 'Country 1',
    price: 100,
    images: ['image1'],
    position: { latitude: 0, longitude: 0 },
    purpose: 'rent',
  },
];

describe('PropertyList ', () => {
  test('Snapshot test', () => {
    const { asFragment } = render(<PropertyList properties={properties} isLoading={false} />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('Should render PropertyList', () => {
    render(<PropertyList properties={properties} isLoading={false} />);
    const PropertyListElement = screen.getByTestId('PropertyList');
    expect(PropertyListElement).toBeInTheDocument();
  });
  test('Should render PropertyList with skeleton', () => {
    render(<PropertyList properties={null} isLoading />);
    const PropertyListElement = screen.getByTestId('PropertyList');
    expect(PropertyListElement).toBeInTheDocument();
  });
  test('Should navigate to PropertyCard', () => {
    render(<PropertyList properties={properties} isLoading={false} />);
    const propertyCard = screen.getAllByTestId('PropertyCard')[0];
    propertyCard.click();
    expect(mockedUsedNavigate).toBeCalled();
  });
});
