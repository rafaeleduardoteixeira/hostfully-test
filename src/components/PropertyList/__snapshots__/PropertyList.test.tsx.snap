// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyList  Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="sc-braxZu fuUgyo"
    data-testid="PropertyList"
  >
    <h1
      class="sc-gJhJTp dOtqgx"
    >
      Properties For Rent
    </h1>
    <div
      class="sc-elDIKY ktHpUm"
    />
    <div
      class="sc-fQpRED guSMlV"
    >
      <div
        class="sc-iCZwEW cIkXSN"
        data-testid="PropertyCard"
      >
        <img
          alt="City 1, Country 1"
          class="sc-knesRu dSuBXv"
          src="image1"
        />
        <div
          class="sc-dnaUSb nKZde"
        >
          <p
            class="sc-kkmypM gUkkno"
          >
            City 1, Country 1
          </p>
          <h3
            class="sc-dkjaqt jcLtgh"
          >
            Property 1
          </h3>
          <p
            class="sc-jCbFiK colLGs"
          >
            $100
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
